pkg/notification/service/fetch.ts

```diff
@@ -0,0 +1,56 @@
+import { Result } from '@mikuroxina/mini-fn';
+import type { AccountID } from '../../accounts/model/account.js';
+import type {
+  Notification,
+  NotificationID,
+} from '../model/notificationBase.js';
+import type {
+  NotificationFilter,
+  NotificationRepository,
+} from '../model/repository/notification.js';
+
+export class FetchNotificationService {
+  constructor(
+    private readonly notificationRepository: NotificationRepository,
+  ) {}
+
+  async fetchByID(
+    id: NotificationID,
+    actorID: AccountID,
+  ): Promise<Result.Result<Error, Notification>> {
+    const res = await this.notificationRepository.findByID(id);
+    if (Result.isErr(res)) {
+      return res;
+    }
+    if (!this.isAllowed(Result.unwrap(res), actorID)) {
+      return Result.err(new Error('not allowed'));
+    }
+
+    return Result.ok(Result.unwrap(res));
```

Comment:
```suggestion
    return res;
```
`unwrap` して `ok` に包み直すのは冗長かと思います. 50 行目も同様です.

Reply:
https://github.com/pulsate-dev/pulsate/pull/1038/commits/a8d2b4c905c8b5b984504218cd173fd28e42c6a1